central/localscanner/certificates_test.go

```diff
@@ -85,6 +86,10 @@ func (s *localScannerSuite) TestCertificateGeneration() {
 			[]string{"scanner.stackrox", "scanner.stackrox.svc", "scanner.namespace", "scanner.namespace.svc"}},
 		{storage.ServiceType_SCANNER_DB_SERVICE, "SCANNER_DB_SERVICE",
 			[]string{"scanner-db.stackrox", "scanner-db.stackrox.svc", "scanner-db.namespace", "scanner-db.namespace.svc"}},
+		{storage.ServiceType_SCANNER_V4_INDEXER_SERVICE, "SCANNER_V4_INDEXER_SERVICE",
+			[]string{"scanner-v4-indexer.stackrox", "scanner-v4-indexer.stackrox.svc", "scanner-v4-indexer.namespace", "scanner-v4-indexer.namespace.svc"}},
+		{storage.ServiceType_SCANNER_V4_DB_SERVICE, "SCANNER_V4_DB_SERVICE",
+			[]string{"scanner-v4-db.stackrox", "scanner-v4-db.stackrox.svc", "scanner-v4-db.namespace", "scanner-v4-db.namespace.svc"}},
```

Comment:
Perhaps a downside to making `serviceTypes` global is it's more difficult to test with the feature both on and off.

These tests are likely to break based on env vars set prior to running the tests.  Would recommend adjusting the tests to either:

- Move the serviceTypes back to non-global and evaluate each time, that will allow for using [`MustUpdatedFeature`](https://github.com/stackrox/stackrox/blob/5ace59b3c465414f264e4b7e2ad98bb21d7ac513/pkg/testutils/feature_flag.go#L19) in the tests to validate behaviors when feature is on or off. 
- Change the tests to have conditions based on the currently set value of `features.ScannerV4.Enabled()`, for example if `ScannerV4` is disabled, then shouldn't expect the `SCANNER_V4` certs to be generated.

Reply:
Actually I have attempted simplify this in a slightly different direction:

While `TestServiceIssueLocalScannerCerts` issues certs depending on `features.ScannerV4.Enabled()` I thought that such a conditional is not really necessary for `generateServiceCertMap`. Instead I have defined a global `allSupportedServiceTypes` which evaluates the unchangeable feature flag `features.ScannerV4Support` and then `allSupportedServiceTypes` is used for validating the cert generation request within `generateServiceCertMap`.

Furthermore I have extended the tests to check the v4-enabled and the v4-disabled case.

I have also extended the test so that it also checks if all expected certificates have in fact been generated. Previously it only tested if the generated certs was a subset of the expected certs.

Reply:
Ohh, I like that better, well done.